src/app/app.component.ts

```diff
@@ -21,4 +26,37 @@ export class AppComponent {
       return 'off-home';
     }
   }
+
+  ngDoCheck() {
+    if (this.auth.authenticated() && this.currentUserId == null)
```

Comment:
It's the only lifecycle I got to make this work. I believe the OnInit is passed thru only once and a subsequent invocation of AppComponent occurs only when login is detected. If you know of another appropriate lifecycle let me know. Otherwise, we can leave this as is for now and I'll come back to it later.